language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/util/ForeachUtil.java

```diff
@@ -0,0 +1,189 @@
+            TypeDescKind.STREAM, TypeDescKind.XML);
+
+    private ForeachUtil() {
+    }
+
+    /**
+     * Given a expression node in a particular completion context,
+     * validate if the expression is an iterable and returns the corresponding foreach completion item.
+     *
+     * @param ctx        completion context
+     * @param expr       expression node
+     * @param typeSymbol type symbol corresponding to the expression node
+     * @return a list of {@link LSCompletionItem}.
+     */
+    public static List<LSCompletionItem> getForeachCompletionItemForIterable(BallerinaCompletionContext ctx,
+                                                                             FieldAccessExpressionNode expr,
+                                                                             Optional<TypeSymbol> typeSymbol) {
+        List<LSCompletionItem> completionItems = new ArrayList<>();
+        if (isInBlockContext(expr) && expr.expression() instanceof SimpleNameReferenceNode) {
+            completionItems.addAll(
+                    typeSymbol.stream().filter(tSymbol -> {
+                                if (tSymbol.typeKind() == TypeDescKind.TYPE_REFERENCE) {
+                                    return ITERABLES.contains(((TypeReferenceTypeSymbol) tSymbol)
+                                            .typeDescriptor().typeKind());
+                                } else {
+                                    return ITERABLES.contains(tSymbol.typeKind());
+                                }
+                            }
+                    ).map(tSymbol -> getForEachCompletionItemForIterable(tSymbol, ctx, expr))
+                            .collect(Collectors.toList()));
+        }
+        return completionItems;
+    }
+
+    /**
+     * Given an iterable field access expression returns the foreach completion item for it.
+     *
+     * @param symbol {@link ArrayTypeSymbol} symbol
+     * @param ctx    completion context
+     * @param expr   expression node
+     * @return {@link LSCompletionItem}
+     */
+    public static LSCompletionItem getForEachCompletionItemForIterable(TypeSymbol symbol,
+                                                                       BallerinaCompletionContext ctx,
+                                                                       FieldAccessExpressionNode expr) {
+        String symbolName = ((SimpleNameReferenceNode) expr.expression()).name().text();
+        String detail = "foreach snippet for iterable variable - " + symbolName;
+        String label = "foreach - " + symbolName;
+        String type = getTypeOfIteratorVariable(symbol);
+        StringBuilder snippet = new StringBuilder("foreach");
+        snippet.append(" ").append(type).append(" ")
+                .append(CommonUtil.getValidatedSymbolName(ctx, VAR_NAME)).append(" in ").append(symbolName)
+                .append(" ").append("{").append(CommonUtil.LINE_SEPARATOR).append("\t${1:}")
```

Comment:
For the placeholder in the snippet, we don't need to have the `:` since we do not have a place holder name.